nanoFramework.Tools.DebugLibrary.Shared/PortSerial/PortSerial.cs

```diff
@@ -282,13 +333,35 @@ public byte[] ReadBuffer(int bytesToRead)
 
                 try
                 {
-                    int readBytes = Device.Read(buffer, 0, bytesToRead);
+                    // this loop shall fix issues with M1 and some drivers showing up bytes to fast
+                    List<byte> received = new List<byte>();
+                    int readBytes = 0;
+
+                    while (Device.BytesToRead > 0 && readBytes < bytesToRead)
+                    {
+                        // compute how many bytes shall be read from the device
+                        // so that we don't read more than what was requested
+                        int missingBytes = Math.Min(Device.BytesToRead, bytesToRead - readBytes);

```

Comment:
Initial code was just reading all the buffer ;-) So definitely a nice improvement here in all cases.